crates/sui-types/src/accumulator_event.rs

```diff
@@ -22,4 +27,57 @@ impl AccumulatorEvent {
             write,
         }
     }
+
+    pub fn total_sui_in_event(&self) -> (u64 /* input */, u64 /* output */) {
+        let Self {
+            write:
+                AccumulatorWriteV1 {
+                    address: AccumulatorAddress { ty, .. },
+                    operation,
+                    value,
+                },
+            ..
+        } = self;
+
+        let TypeTag::Struct(struct_ty) = ty else {
+            return (0, 0);
+        };
+
+        if !(struct_ty.address == SUI_FRAMEWORK_ADDRESS
+            && struct_ty.module.as_str() == "balance"
+            && struct_ty.name.as_str() == "Balance")
```

Comment:
Also we might have a resolved type here which would be more canonical (it gives a triple of address+module+datatype)